### PR TITLE
fix!: ESM, drop API, drop Node 12 support

### PR DESCRIPTION
**Pull request by alexander-fenster**
This package is quite unique compared to other things in `googleapis` repository. It's a tool and not a library, and it has some dependencies we don't normally mess with, such as `chalk`, `ora`, and `meow`.

Since most of the dependencies dropped Node 12 in their latest versions and became ESM friendly, I'm suggesting this refactor to make this tool ESM compatible. I used `ncu -u` to bump all the dependencies to their latest versions, and make some code fixes to make it all work together.

**BREAKING CHANGE**: originally this tool was created (by me!) as a library, providing some API, with an expectation that the users will actually write the code to maintain GitHub repositories. It's not the case anymore, it was converted to a CLI tool long ago, and the main use case now is just use the `repo` CLI. I'm dropping parts of the API previously exported and the samples showing how to use it, because I don't want to keep that bad code (also, because its tests used `proxyquire`, which does not play well with ESM).

**BREAKING CHANGE**: drop Node 12 support.

Other than that, everything should just work!

### CODE REVIEW

File: src/list-issues.ts [hunk 1]
```diff
@@ -13,10 +13,9 @@
 // limitations under the License.
 
 import * as meow from 'meow';
-import {meowFlags} from './cli';
 
-import {GitHubRepository, Issue} from './lib/github';
-import {processIssues} from './lib/asyncItemIterator';
+import {GitHubRepository, Issue} from './lib/github.js';
+import {processIssues} from './lib/asyncItemIterator.js';
```
Comment (chingor13) [inline src/list-issues.ts:18]: I'm not very knowledgeable in ESM/CJS stuff, but it seems weird to need the `.js`?

**Review by chingor13 — COMMENTED** [review]
For the most part this looks ok to me.

It looks like the only functionality being removing is the "update single file" functionality.

File: src/list-issues.ts (hunk 1 above)
Comment (alexander-fenster) [reply src/list-issues.ts:18]: Weird indeed, but it cannot resolve imports without it. https://nodejs.org/api/esm.html#esm_mandatory_file_extensions

TypeScript understands these imports during compilation and preserves them, so they turn out just fine in the `build` folder.

**Comment by alexander-fenster** [comment]
> For the most part this looks ok to me.
> 
> It looks like the only functionality being removing is the "update single file" functionality.

And something about updating in branches, which I don't even remember what it was doing.

Technically, I could leave those APIs, but I need to remove their unit tests (since they use `proxyquire` which cannot be used with ESM), and since the APIs are unused, I decided it's better to remove everything unused rather than keep supporting dead code not even covered by tests.

**Comment by alexander-fenster** [comment]
@sofisl @bcoe @chingor13 ping on this :)

**Review by sofisl — APPROVED** [review]

**Pull request merged**